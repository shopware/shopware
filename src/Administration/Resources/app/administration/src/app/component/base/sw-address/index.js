import template from './sw-address.html.twig';
import './sw-address.scss';

const { Component } = Shopware;

/**
 * @public
 * @description Component to render a postal address
 * @status ready
 * @example-type static
 * @component-example
 * <sw-address headline="Billing address" :address="{
 *     salutation: 'Mister',
 *     title: 'Doctor',
 *     firstName: 'John',
 *     lastName: 'Doe',
 *     street: 'Main St 123',
 *     zipcode: '12456',
 *     city: 'Anytown',
 *     country: { name: 'Germany' }
 * }"></sw-address>
 */
Component.register('sw-address', {
    template,

    props: {
        address: {
            type: Object,
            default() {
                return {
                    salutation: '',
                    title: '',
                    firstName: '',
                    lastName: '',
                    street: '',
                    zipcode: '',
                    city: '',
                    country: {
                        name: '',
                    },
                    countryState: {
                        name: '',
                    },
                };
            },
        },

        headline: {
            type: String,
            required: false,
            default: '',
        },

        showEditButton: {
            type: Boolean,
            required: false,
            default: false,
        },

        editLink: {
            type: Object,
            required: false,
            default: null,
        },
    },

    computed: {
        addressClasses() {
            return {
                'sw-address--headline': this.headline,
            };
        },
    },
});
