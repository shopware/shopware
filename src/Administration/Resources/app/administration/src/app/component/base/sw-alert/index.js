import template from './sw-alert.html.twig';
import './sw-alert.scss';

const { Component } = Shopware;

/**
 * @description
 * The <u>sw-alert</u> component is used to convey important information to the user. It comes in 4 variations,
 * <strong>success</strong>, <strong>info</strong>, <strong>warning</strong> and <strong>error</strong>. These have
 * default icons assigned which can be changed and represent different actions
 * @status ready
 * @example-type dynamic
 * @component-example
 * <sw-alert variant="info" title="Example title" :closable="true">
 *    Sample text
 * </sw-alert>
 */
Component.register('sw-alert', {
    template,

    props: {
        variant: {
            type: String,
            default: 'info',
            validValues: ['info', 'warning', 'error', 'success'],
            validator(value) {
                return ['info', 'warning', 'error', 'success'].includes(value);
            }
        },
        appearance: {
            type: String,
            default: 'default',
            validValues: ['default', 'notification', 'system'],
            validator(value) {
                return ['default', 'notification', 'system'].includes(value);
            }
        },
        title: {
            type: String,
            required: false,
            default: ''
        },
        showIcon: {
            type: Boolean,
            required: false,
            default: true
        },
        closable: {
            type: Boolean,
            required: false,
            default: false
        },
        notificationIndex: {
            type: String,
            required: false
        }
    },
    computed: {
        alertClasses() {
            return [
                `sw-alert--${this.variant}`,
                `sw-alert--${this.appearance}`,
                {
                    'sw-alert--no-icon': !this.showIcon,
                    'sw-alert--closable': this.closable,
                    'sw-alert--actions': this.hasActionSlot
                }
            ];
        },

        alertIcon() {
            const iconConfig = {
                info: 'default-badge-info',
                warning: 'default-badge-warning',
                error: 'default-badge-error',
                success: 'default-basic-checkmark-circle'
            };

            return iconConfig[this.variant] || 'default-bell-bell';
        },

        hasActionSlot() {
            return !!this.$slots.actions;
        }
    }
});
