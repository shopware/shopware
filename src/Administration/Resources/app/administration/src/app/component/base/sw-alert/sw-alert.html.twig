{% block sw_alert %}
<div
    class="sw-alert"
    :class="alertClasses"
    role="alert"
>
    {% block sw_alert_inner %}
    {% block sw_alert_icon %}
    <slot name="customIcon">
        <sw-icon
            v-if="showIcon"
            size="20px"
            class="sw-alert__icon"
            :name="alertIcon"
            decorative
        />
    </slot>
    {% endblock %}

    {% block sw_alert_body %}
    <div
        class="sw-alert__body"
        :class="alertBodyClasses"
    >
        {% block sw_alert_title %}
        <div
            v-if="title"
            class="sw-alert__title"
        >
            {{ title }}
        </div>
        {% endblock %}

        {% block sw_alert_message %}
        <div class="sw-alert__message">
            <slot>{% block sw_alert_slot_default %}{% endblock %}</slot>
        </div>
        {% endblock %}
    </div>
    {% endblock %}

    {% block sw_alert_actions %}
    <div
        v-if="hasActionSlot"
        class="sw-alert__actions"
    >
        <slot name="actions"></slot>
    </div>
    {% endblock %}

    {% block sw_alert_close %}
    <button
        v-if="closable"
        class="sw-alert__close"
        aria-label="Schließen"
        title="Schließen"
        @click.prevent="$emit('close', notificationIndex)"
    >
        <sw-icon
            name="small-default-x-line-medium"
            small
        />
    </button>
    {% endblock %}
    {% endblock %}
</div>
{% endblock %}
