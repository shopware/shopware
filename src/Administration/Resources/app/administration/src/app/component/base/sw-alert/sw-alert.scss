@import "~scss/variables";

$sw-alert-box-shadow: 1px 2px 5px rgba(0, 0, 0, 0.1);
$sw-alert-size-close: 40px;

.sw-alert {
    border: 1px solid $color-gray-300;
    border-radius: $border-radius-default;
    color: $color-darkgray-200;
    background-color: $color-white;
    text-align: left;
    position: relative;
    margin: 0 auto 20px;
    font-size: $font-size-small;

    .sw-alert__body {
        padding: 15px 15px 15px 55px;
    }

    &.sw-alert--no-icon {
        padding: 15px 30px;
    }

    .sw-alert__icon {
        color: $color-gray-300;
        position: absolute;
        display: block;
        left: 15px;
        top: 15px;
    }

    .sw-alert__close {
        width: $sw-alert-size-close;
        height: $sw-alert-size-close;
        line-height: $sw-alert-size-close;
        position: absolute;
        display: block;
        top: 0;
        right: 0;
        padding: 0;
        margin: 0;
        text-align: center;
        background: none;
        border: 0 none;
        outline: none;
        cursor: pointer;
    }

    .sw-alert__title {
        font-size: $font-size-small;
        margin-bottom: 8px;
        display: block;
        font-weight: bold;
    }

    .sw-alert__message {
        font-size: $font-size-small;
    }

    .sw-alert__actions {
        border-top: 1px solid $color-gray-300;
        display: flex;
        flex-wrap: wrap;

        .sw-button {
            border: 0;
            height: 40px;
            background: none;
            border-radius: 0;
            flex-basis: 0;
            flex-grow: 1;
            max-width: 100%;
            font-weight: 500;

            &:not(:last-child) {
                border-right: 1px solid $color-gray-300;
            }

            &:hover {
                background-color: $color-white;
            }
        }
    }

    &.sw-alert--closable {
        .sw-alert__body {
            padding-right: $sw-alert-size-close;
        }
    }

    &.sw-alert--notification {
        box-shadow: $sw-alert-box-shadow;
        border: 1px solid $color-gray-300;
        border-left: 4px solid $color-gray-300;

        .sw-alert__actions .sw-button {
            color: $color-shopware-brand-500;
        }
    }

    &.sw-alert--warning {
        border-color: $color-pumpkin-spice-500;
        background-color: $color-pumpkin-spice-50;

        &.sw-alert--notification {
            background-color: $color-white;
            border-color: $color-gray-300;
            border-left-color: $color-pumpkin-spice-500;
        }

        .sw-alert__icon,
        .sw-alert__close {
            color: $color-pumpkin-spice-500;
        }

        .sw-alert__actions {
            border-color: $color-pumpkin-spice-200;

            .sw-button {
                border-color: $color-pumpkin-spice-200;

                &:hover {
                    background-color: $color-pumpkin-spice-100;
                }
            }
        }
    }

    &.sw-alert--error {
        border-color: $color-crimson-500;
        background-color: $color-crimson-50;

        &.sw-alert--notification {
            background-color: $color-white;
            border-color: $color-gray-300;
            border-left-color: $color-crimson-500;
        }

        .sw-alert__icon,
        .sw-alert__close {
            color: $color-crimson-500;
        }

        .sw-alert__actions {
            border-color: $color-crimson-200;

            .sw-button {
                border-color: $color-crimson-200;

                &:hover {
                    background-color: $color-crimson-100;
                }
            }
        }
    }

    &.sw-alert--success {
        border-color: $color-emerald-500;
        background-color: $color-emerald-50;

        &.sw-alert--notification {
            background-color: $color-white;
            border-color: $color-gray-300;
            border-left-color: $color-emerald-500;
        }

        .sw-alert__icon,
        .sw-alert__close {
            color: $color-emerald-500;
        }

        .sw-alert__actions {
            border-color: $color-emerald-200;

            .sw-button {
                border-color: $color-emerald-200;

                &:hover {
                    background-color: $color-emerald-100;
                }
            }
        }
    }

    &.sw-alert--info {
        border-color: $color-shopware-brand-500;
        background-color: $color-shopware-brand-50;

        &.sw-alert--notification {
            background-color: $color-white;
            border-color: $color-gray-300;
            border-left-color: $color-shopware-brand-500;
        }

        .sw-alert__icon,
        .sw-alert__close {
            color: $color-shopware-brand-500;
        }

        .sw-alert__actions {
            border-color: $color-shopware-brand-200;

            .sw-button {
                border-color: $color-shopware-brand-200;

                &:hover {
                    background-color: $color-shopware-brand-100;
                }
            }
        }
    }

    &.sw-alert--notification {
        background-color: $color-gray-100;

        .sw-alert__actions {
            border-color: $color-gray-300;

            .sw-button {
                border-color: $color-gray-300;

                &:hover {
                    background-color: $color-gray-100;
                }
            }
        }
    }

    &.sw-alert--system {
        color: $color-white;
        background: $color-deep-cove;
        border: 1px solid $color-darkgray-200;
        box-shadow: none;

        .sw-alert__actions {
            border-color: $color-darkgray-50;

            .sw-button {
                color: $color-gray-300;
                border-color: $color-darkgray-50;

                &:hover {
                    background-color: lighten($color-deep-cove, 6%);
                }

                &:active {
                    background-color: lighten($color-deep-cove, 8%);
                }
            }
        }
    }
}
