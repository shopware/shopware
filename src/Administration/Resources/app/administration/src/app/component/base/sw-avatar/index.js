import template from './sw-avatar.html.twig';
import './sw-avatar.scss';

const { Component } = Shopware;
const { cloneDeep } = Shopware.Utils.object;

const colors = [
    '#FFD700',
    '#FFC700',
    '#F88962',
    '#F56C46',
    '#FF85C2',
    '#FF68AC',
    '#6AD6F0',
    '#4DC6E9',
    '#A092F0',
    '#8475E9',
    '#57D9A3',
    '#3CCA88',
];

/**
 * @description The component helps adding a custom user image or initials to the administration.
 * @status ready
 * @example-type static
 * @component-example
 * <div style="display: flex; align-items: center;">
 * <sw-avatar color="#dd4800"
 *            size="48px"
 *            firstName="John"
 *            style="margin: 0 10px;"
 *            lastName="Doe"></sw-avatar>
 *
 * <sw-avatar size="48px"
 *            imageUrl="https://randomuser.me/api/portraits/women/68.jpg"></sw-avatar>
 * </div>
 *
 * <sw-avatar size="48px"
 *            imageUrl="https://randomuser.me/api/portraits/men/68.jpg"
 *            :sourceContext="user"></sw-avatar>
 * </div>
 */
Component.register('sw-avatar', {
    template,

    props: {
        color: {
            type: String,
            required: false,
            default: '',
        },
        size: {
            type: String,
            required: false,
            default: null,
        },
        firstName: {
            type: String,
            required: false,
            default: '',
        },
        lastName: {
            type: String,
            required: false,
            default: '',
        },
        imageUrl: {
            type: String,
            required: false,
            default: null,
        },
        placeholder: {
            type: Boolean,
            required: false,
            default: false,
        },
        sourceContext: {
            type: Object,
            required: false,
            default: null,
        },
        variant: {
            type: String,
            required: false,
            default: 'circle',
            validator: (value) => {
                return ['circle', 'square'].includes(value);
            },
        },
    },

    data() {
        return {
            fontSize: 16,
            lineHeight: 16,
        };
    },

    computed: {
        avatarSize() {
            const size = this.size;

            return {
                width: size,
                height: size,
            };
        },

        avatarInitials() {
            const firstNameLetter = this.firstName ? this.firstName[0] : '';
            const lastNameLetter = this.lastName ? this.lastName[0] : '';

            return firstNameLetter + lastNameLetter;
        },

        avatarInitialsSize() {
            return {
                'font-size': `${this.fontSize}px`,
                'line-height': `${this.lineHeight}px`,
            };
        },

        avatarImage() {
            if (this.imageUrl) {
                return { 'background-image': `url('${this.imageUrl}')` };
            }

            if (!this.sourceContext || !this.sourceContext.avatarMedia || !this.sourceContext.avatarMedia.url) {
                return null;
            }

            const avatarMedia = cloneDeep(this.sourceContext.avatarMedia);
            const thumbnailImage = avatarMedia.thumbnails.sort((a, b) => a.width - b.width)[0];
            const previewImageUrl = thumbnailImage ? thumbnailImage.url : avatarMedia.url;

            return { 'background-image': `url('${previewImageUrl}')` };
        },

        avatarColor() {
            if (this.color.length) {
                return {
                    'background-color': this.color,
                };
            }

            const firstNameLength = this.firstName ? this.firstName.length : 0;
            const lastNameLength = this.lastName ? this.lastName.length : 0;

            const nameLength = firstNameLength + lastNameLength;
            const color = colors[nameLength % colors.length];

            return {
                'background-color': color,
            };
        },

        hasAvatarImage() {
            return !!this.avatarImage && !!this.avatarImage['background-image'];
        },

        showPlaceholder() {
            return this.placeholder && !this.hasAvatarImage;
        },

        showInitials() {
            return !this.placeholder && !this.hasAvatarImage;
        },
    },

    watch: {
        size() {
            this.$nextTick(() => {
                this.generateAvatarInitialsSize();
            });
        },
    },

    mounted() {
        this.mountedComponent();
    },

    methods: {
        mountedComponent() {
            this.generateAvatarInitialsSize();
        },

        generateAvatarInitialsSize() {
            const avatarSize = this.$refs.swAvatar.offsetHeight;

            this.fontSize = Math.round(avatarSize * 0.4);
            this.lineHeight = Math.round(avatarSize * 0.98);
        },
    },
});
