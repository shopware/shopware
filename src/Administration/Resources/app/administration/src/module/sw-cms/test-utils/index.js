import { mount } from '@vue/test-utils';

function runGenericCmsTest(component) {
    beforeAll(async () => {
        await setupCmsEnvironment();
    });

    it('should be a Vue.js component', async () => {
        const wrapper = mount(component);
        expect(wrapper.vm).toBeTruthy();
    });
}

function runCmsBlockRegistryTest(config) {
    beforeAll(async () => {
        await setupCmsEnvironment();
        await import(config.import);
    });

    it('registers the component in the CMS block registry', async () => {
        const componentRegistry = Shopware.Component.getComponentRegistry();
        const cmsService = Shopware.Service('cmsService');

        expect(componentRegistry.has(config.component)).toBeTruthy();
        expect(componentRegistry.has(config.preview)).toBeTruthy();

        const blockConfig = cmsService.getCmsBlockConfigByName(config.name);

        expect(blockConfig.component).toBe(config.component);
        expect(blockConfig.previewComponent).toBe(config.preview);
    });
}

function runCmsElementRegistryTest(config) {
    beforeAll(async () => {
        await setupCmsEnvironment();
        await import(config.import);
    });

    it('registers the component in the CMS block registry', async () => {
        const componentRegistry = Shopware.Component.getComponentRegistry();
        const cmsService = Shopware.Service('cmsService');

        expect(componentRegistry.has(config.component)).toBeTruthy();

        if (config.preview) {
            expect(componentRegistry.has(config.preview)).toBeTruthy();
        }

        const elementConfig = cmsService.getCmsElementConfigByName(config.name);

        expect(elementConfig.component).toBe(config.component);
        expect(elementConfig.configComponent).toBe(config.config);

        if (config.preview) {
            expect(elementConfig.previewComponent).toBe(config.preview);
        }
    });
}

async function setupCmsEnvironment() {
    await import('src/module/sw-cms/store/cms-page.store');
    await import('src/module/sw-cms/service/cms.service');
    await import('src/module/sw-cms/service/cms-element-favorites.service');
    await import('src/module/sw-cms/mixin/sw-cms-element.mixin');
    await import('src/module/sw-cms/mixin/sw-cms-state.mixin');

    Shopware.State.get('session').currentUser = {
        id: 'admin',
    };
}

/**
 * @private
 * @package buyers-experience
 */
export { runGenericCmsTest, runCmsBlockRegistryTest, runCmsElementRegistryTest, setupCmsEnvironment };
