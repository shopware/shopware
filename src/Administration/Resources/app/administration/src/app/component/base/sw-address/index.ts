import type { PropType } from 'vue';
import type { Route } from 'vue-router';
import type { Address } from 'src/core/service/api/custom-snippet.api.service';
import template from './sw-address.html.twig';
import './sw-address.scss';

const { Component } = Shopware;

/**
 * @package admin
 *
 * @deprecated tag:v6.6.0 - Will be private
 * @public
 * @description Component to render a postal address
 * @status ready
 * @example-type static
 * @component-example
 * <sw-address headline="Billing address" :address="{
 *     salutation: 'Mister',
 *     title: 'Doctor',
 *     firstName: 'John',
 *     lastName: 'Doe',
 *     street: 'Main St 123',
 *     zipcode: '12456',
 *     city: 'Anytown',
 *     country: { name: 'Germany' }
 * }" :formattingAddress="First Name Last Name\nGermany"></sw-address>
 */
Component.register('sw-address', {
    template,

    props: {
        address: {
            type: Object as PropType<Address>,
            default(): Address {
                return {
                    salutation: {
                        displayName: '',
                        translated: {
                            displayName: '',
                        },
                    },
                    title: '',
                    firstName: '',
                    lastName: '',
                    street: '',
                    zipcode: '',
                    city: '',
                    country: {
                        name: '',
                        translated: {
                            name: '',
                        },
                    },
                    countryState: {
                        name: '',
                        translated: {
                            name: '',
                        },
                    },
                };
            },
        },

        headline: {
            type: String,
            required: false,
            default: '',
        },

        formattingAddress: {
            type: String,
            required: false,
            default: null,
        },

        showEditButton: {
            type: Boolean,
            required: false,
            default: false,
        },

        editLink: {
            type: Object as PropType<Route | null>,
            required: false,
            default: null,
        },
    },

    computed: {
        addressClasses(): Record<string, boolean | string> {
            return {
                'sw-address--headline': this.headline,
            };
        },

        displayFormattingAddress(): string {
            return this.formattingAddress;
        },
    },
});
