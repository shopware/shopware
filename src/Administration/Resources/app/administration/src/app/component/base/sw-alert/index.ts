import type { NotificationType } from 'src/app/mixin/notification.mixin';
import type { PropType } from 'vue';
import template from './sw-alert.html.twig';
import './sw-alert.scss';

const { Component } = Shopware;
type AppearanceType = 'default' | 'notification' | 'system';
type CssClassesObject = { [key: string]: boolean };
type CssClasses = Array<string | CssClassesObject> | CssClassesObject;

/**
 * @package admin
 *
 * @deprecated tag:v6.6.0 - Will be private
 * @description
 * The <u>sw-alert</u> component is used to convey important information to the user. It comes in 4 variations,
 * <strong>success</strong>, <strong>info</strong>, <strong>warning</strong> and <strong>error</strong>. These have
 * default icons assigned which can be changed and represent different actions
 * @status ready
 * @example-type dynamic
 * @component-example
 * <sw-alert variant="info" title="Example title" :closable="true">
 *    Sample text
 * </sw-alert>
 */
Component.register('sw-alert', {
    template,

    props: {
        variant: {
            type: String as PropType<NotificationType>,
            required: false,
            default: 'info',
            validValues: ['info', 'warning', 'error', 'success', 'neutral'],
            validator(value: string): boolean {
                return ['info', 'warning', 'error', 'success', 'neutral'].includes(value);
            },
        },
        appearance: {
            type: String as PropType<AppearanceType>,
            required: false,
            default: 'default',
            validValues: ['default', 'notification', 'system'],
            validator(value: string) {
                return ['default', 'notification', 'system'].includes(value);
            },
        },
        title: {
            type: String,
            required: false,
            default: '',
        },
        showIcon: {
            type: Boolean,
            required: false,
            // TODO: Boolean props should only be opt in and therefore default to false
            // eslint-disable-next-line vue/no-boolean-default
            default: true,
        },
        closable: {
            type: Boolean,
            required: false,
            default: false,
        },
        notificationIndex: {
            type: String,
            required: false,
            default: null,
        },
        icon: {
            type: String,
            required: false,
            default: null,
        },
    },
    computed: {
        alertIcon(): string {
            if (this.icon) {
                return this.icon;
            }

            const iconConfig: { [type: string]: string } = {
                info: 'regular-info-circle',
                warning: 'regular-exclamation-triangle',
                error: 'regular-exclamation-circle',
                success: 'regular-check-circle',
                neutral: 'regular-info-circle',
            };

            return iconConfig[this.variant] || 'regular-bell';
        },

        hasActionSlot(): boolean {
            return !!this.$slots.actions;
        },

        alertClasses(): CssClasses {
            const variantClass = `sw-alert--${this.variant}`;

            return [
                `sw-alert--${this.appearance}`,
                {
                    [variantClass]: this.appearance !== 'system',
                    'sw-alert--icon': this.showIcon,
                    'sw-alert--no-icon': !this.showIcon,
                    'sw-alert--closable': this.closable,
                    'sw-alert--actions': this.hasActionSlot,
                },
            ];
        },

        alertBodyClasses(): CssClasses {
            return {
                'sw-alert__body--icon': this.showIcon,
                'sw-alert__body--closable': this.closable,
            };
        },
    },
});
