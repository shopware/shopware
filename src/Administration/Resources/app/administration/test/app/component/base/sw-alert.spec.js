import { shallowMount } from '@vue/test-utils';
import 'src/app/component/base/sw-alert';

describe('components/base/sw-alert', () => {
    let wrapper;

    afterEach(() => { if (wrapper) wrapper.destroy(); });

    it('should be a Vue.js component', async () => {
        wrapper = shallowMount(Shopware.Component.build('sw-alert'), {
            stubs: ['sw-icon']
        });
        expect(wrapper.vm).toBeTruthy();
    });

    it('should render correctly', async () => {
        const title = 'Alert title';
        const message = '<p>Alert message</p>';

        wrapper = shallowMount(Shopware.Component.build('sw-alert'), {
            stubs: ['sw-icon'],
            propsData: {
                title
            },
            slots: {
                default: message
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it('should use custom icon', async () => {
        wrapper = shallowMount(Shopware.Component.build('sw-alert'), {
            stubs: ['sw-icon'],
            propsData: {
                icon: 'your-icon-here'
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});

