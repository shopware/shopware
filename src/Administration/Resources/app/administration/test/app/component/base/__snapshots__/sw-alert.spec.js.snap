// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/base/sw-alert should render correctly 1`] = `
<div
  class="sw-alert sw-alert--info sw-alert--default sw-alert--icon"
  role="alert"
>
  <sw-icon-stub
    class="sw-alert__icon"
    decorative=""
    name="default-badge-info"
    size="20px"
  />
   
  <div
    class="sw-alert__body sw-alert__body--icon"
  >
    <div
      class="sw-alert__title"
    >
      
            Alert title
        
    </div>
     
    <div
      class="sw-alert__message"
    >
      <p>
        Alert message
      </p>
    </div>
  </div>
   
  <!---->
   
  <!---->
</div>
`;
